### PR TITLE
add re-rendering to the calendar when props change

### PR DESCRIPTION
**Pull request by mixcas**

### CODE REVIEW

File: src/native/components/calendar/Calendar.js [hunk 1]
```diff
@@ -116,6 +116,8 @@ class Calendar extends Component {
     // Empty Calendar
     if(this.dates.length === 0) return null;
 
+    this.dates = getDates(activeCatalogo, futureCatalogos);
```
Comment (varffs) [inline src/native/components/calendar/Calendar.js:119]: should this happen in `componentWillReceiveProps`?

Comment (mixcas) [reply src/native/components/calendar/Calendar.js:119]: No, `activeCatalogo` and `futureCatalogos` come from props so a change in those values trigger rerender. The problem was that we were only calling `this.dates = getDates()` on the constructor. So when props changed `this.dates` wasn't updating.

Comment (varffs) [reply src/native/components/calendar/Calendar.js:119]: yh but couldn't we use that event rather then putting this logic here in the `render`?

Comment (mixcas) [reply src/native/components/calendar/Calendar.js:119]: Is not what `componentWillReceiveProps` is for. Also is soon to be [depreacated](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops)

**Pull request merged**